src/PrestaShopBundle/Translation/Provider/DefaultCatalogueProvider.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Translation\Provider;
+
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * Gets catalogue within the files filtered by name in the directory given.
+ */
+class DefaultCatalogueProvider implements TranslationCatalogueProviderInterface
+{
+    /**
+     * @var string
+     */
+    private $directory;
+
+    /**
+     * @var array
+     */
+    private $filenameFilters;
+
+    /**
+     * @param string $directory Directory where to look files
+     * @param string[] $filenameFilters Array of globs to use to match files
+     */
+    public function __construct(string $directory, array $filenameFilters)
+    {
+        $this->directory = $directory;
```

Comment:
```suggestion
        if (!file_exists($directory)) {
            throw new InvalidArgumentException(sprintf('Directory %s does not exist', $directory));
        }

        $this->directory = $directory;
```

Defensive programming 👀  don't trust the developer! 😛

Reply:
:D :D Yeah, never trust them.
But getCatalogueFromPaths checks the directory existence.
As this class will not be used to create a service but instantiated on the fly, I can do the check in the constructor indeed ;) thanks